tests/test_websockets.py

```diff
@@ -391,3 +391,39 @@ async def mock_send(message):
     assert websocket == websocket
     assert websocket in {websocket}
     assert {websocket} == {websocket}
+
+
+def test_websocket_close_reason(test_client_factory) -> None:
+    def app(scope):
+        async def asgi(receive, send):
+            websocket = WebSocket(scope, receive=receive, send=send)
+            await websocket.accept()
+            await websocket.close(code=1001, reason="Closing")
+
+        return asgi
+
+    client = test_client_factory(app)
+    with client.websocket_connect("/") as websocket:
+        with pytest.raises(WebSocketDisconnect) as exc:
+            websocket.receive_text()
+        assert exc.value.code == status.WS_1001_GOING_AWAY
+        assert exc.value.reason == "Closing"
+
+
+def test_websocket_close_reason_manual(test_client_factory) -> None:
+    def app(scope):
+        async def asgi(receive, send):
+            websocket = WebSocket(scope, receive=receive, send=send)
+            await websocket.accept()
+
+            websocket_close = WebSocketClose(code=1001, reason="Closing")
```

Comment:
This test is different from the above test, as that calls `close` on websocket but this one will await `WebSocketClose` manually.

Reply:
But this should probably be tested elsewhere?

Reply:
I don't think we need the standalone anymore, thanks to your suggestion. I say let's drop it when there's no value in it.

Reply:
The only place `WebSocketClose` is used is in the `routing` where we close the connection. And in those cases we rely on the defaults.